src/api/middlewares/requireAuth.ts

```diff
@@ -1,11 +1,10 @@
 import { FastifyReply, FastifyRequest } from 'fastify'
+import { validateSession } from '../../security/session'
 
 export default async (req: FastifyRequest, res: FastifyReply, done: (err?: any) => void): Promise<void> => {
   const hasHeader = !!req.headers.authorization
 
-  // TODO: Validate token authenticity
-
-  if (!hasHeader /* || !tokenIsValid(req.headers.authorization) */) {
+  if (!hasHeader || (await validateSession(req.headers.authorization as string)).isValid) {
     done(new Error('Authorization header not present'))
   }
```

Comment:
Error message is incorrect here, will now misleadingly state that auth header is not present when it is. Negation is furthermore missing, meaning a valid token will result in an error.

Split these into separate branches.
```suggestion
  if (!hasHeader) {
    done(new Error('Authorization header not present'))
  } else if (!(await validateSession(req.headers.authorization)).isValid) {
    done(new Error('Unauthorized')
  }
```